src/HotChocolate/Core/src/Execution/Pipeline/OperationComplexityMiddleware.cs

```diff
@@ -115,17 +116,14 @@ public async ValueTask InvokeAsync(IRequestContext context)
 
             foreach (var analyzer in analyzers)
             {
-                if (analyzer.OperationDefinitionNode == operationDefinition)
+                if (analyzer.OperationDefinitionNode.Equals(operationDefinition, SyntaxComparison.Syntax))
```

Comment:
A reference check should be sufficient here as we are dealing with the same document. Why would you need to compare the syntax tree?